test/functional/lib/tests.cpp

```diff
@@ -2700,14 +2643,41 @@ LoadApiTest()
     const XDP_API_PROVIDER_DISPATCH *ProviderDispatch;
     const XDP_API_PROVIDER_BINDING_CONTEXT *ProviderBindingContext;
 
-    NTSTATUS Status = InitializeApi(&XdpApiClient, &ProviderDispatch, &ProviderBindingContext, &XdpFuncXdpApiClientDispatch, XDP_API_VERSION_1);
+    NTSTATUS Status =
+        XdpOpenApi(
+            XDP_API_VERSION_1, NULL, NULL, &TestDetach, &XdpFuncXdpApiClientDispatch,
```

Comment:
ideally we should do something that verifies the actual API version matches the requested version - i think this is currently not actually working ,based on apikernel.c.

Reply:
How should we access the API version from user side? should we add getter api?
How does user mode xdp support the feature?

Reply:
i think the caller of XdpOpenApi will know the version matches by this design, since it passes it as a parameter.

XDP itself should be able to determine the version based on the NMR provider binding info

Reply:
You mean the validation should not be part of tests, but in client API?

Reply:
And there is actually no V2 api defined in apikernel.c

Reply:
Updated, here checks the version.

https://github.com/microsoft/xdp-for-windows/pull/713/commits/03463f5c4bda5ad0d0d4a0045ea6c270f65ec577#diff-cfb1ad3cc8f2c190e4ada2e79e405ef68e3fb23cdb5fd37bcd656eaada853d52L323-L324